kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -586,6 +587,34 @@ protected S3SinkConnectorConfig(ConfigDef configDef, Map<String, String> props)
     addToGlobal(partitionerConfig);
     addToGlobal(commonConfig);
     addToGlobal(this);
+    validateTimezone();
+  }
+
+  private void validateTimezone() {
+    String timezone = getString(PartitionerConfig.TIMEZONE_CONFIG);
+    if (!timezone.isEmpty()) {
+      try {
+        DateTimeZone.forID(timezone);
+      } catch (IllegalArgumentException e) {
+        throw new ConfigException(
+            String.format("%s configuration invalid: %s",
+                PartitionerConfig.TIMEZONE_CONFIG,
+                e.getMessage()
+            )
+        );
+      }
+    }
```

Comment:
make this an actual validator like `RegionValidator` that validates the actual config